server/tm_cmds.go

```diff
@@ -123,3 +127,67 @@ func VersionCmd() *cobra.Command {
 		},
 	}
 }
+
+// makeKeyMigrateCmd is ported from tendermint's key-migrate command, but
+// uses the SDK's own server.Context.
+// ref: https://github.com/tendermint/tendermint/blob/master/UPGRADING.md#database-key-format-changes
+func makeKeyMigrateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "key-migrate",
+		Short: "Run Tendermint database key migration",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancel := context.WithCancel(cmd.Context())
+			defer cancel()
+
+			serverCtx := GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			contexts := []string{
```

Comment:
I'm shocked we have to actually write any substantial code here. I would've presumed Tendermint would've exposed a command that'll do all of this for us?

In any case, does Tendermint not expose the contexts we want to migrate?

Reply:
TM exposes:

https://github.com/tendermint/tendermint/blob/f33722b4233159a31cdf6c33258fbc601cdbd1d4/cmd/tendermint/commands/key_migrate.go#L15

But it needs additional flags to get the DB dir right. In this PR, we use the same code, but with `serverCtx`

Reply:
I see. Such an easy problem to solve if the TM command just had a function that the command called instead. Sigh...